### PR TITLE
Add Distcc

### PR DESCRIPTION
**Pull request by satmandu**
- Also updates `avahi` and adds back `i686` binaries, and fixes an issue with `dbus` startup showing an error message.
- fix `crew` invoking `crew install` in a subshell.
- Updates `wayland_protocols` => 1.34, `libva`,  `zstd` => 1.5.6, `mesonbuild`, `llvm18_*` => 18.1.2, `cmake` => 3.29.0
- Enable `cmake` 3.29 supported linker properties in `const.rb`.

Works properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` 

### Run the following to get this pull request's changes locally for testing.
```bash
CREW_REPO=https://github.com/satmandu/chromebrew.git CREW_BRANCH=distcc crew update ; yes | crew upgrade
```

### CODE REVIEW

**Comment by satmandu** [comment]
Thinking I'm going to just redo this logic to look at `/usr/local/etc/default/distcc` to see whether distccd should be started.

File: manifest/armv7l/a/avahi.filelist [hunk 1]
```diff
@@ -1,21 +1,9 @@
-/usr/local/bin/avahi-browse
-/usr/local/bin/avahi-browse-domains
-/usr/local/bin/avahi-publish
-/usr/local/bin/avahi-publish-address
-/usr/local/bin/avahi-publish-service
-/usr/local/bin/avahi-resolve
-/usr/local/bin/avahi-resolve-address
-/usr/local/bin/avahi-resolve-host-name
-/usr/local/bin/avahi-set-host-name
```
Comment (uberhacker) [inline manifest/armv7l/a/avahi.filelist:9]: What happened to all the binaries?

File: packages/dbus.rb [hunk 2]
```diff
@@ -1,19 +1,19 @@
-require 'package'
+require 'buildsystems/meson'
 
-class Dbus < Package
+class Dbus < Meson
   description 'D-Bus is a message bus system, a simple way for applications to talk to one another.'
   homepage 'https://www.freedesktop.org/wiki/Software/dbus/'
-  version '1.15.4'
+  version '1.15.8'
```
Comment (uberhacker) [inline packages/dbus.rb:6]: No filelist updates?

File: manifest/armv7l/a/avahi.filelist (hunk 1 above)
Comment (satmandu) [reply manifest/armv7l/a/avahi.filelist:9]: All the gtk requiring binaries have been removed in service of making avahi work w/o a gui.

**Comment by satmandu** [comment]
Still trying to figure out clang issues...

File: bin/crew [hunk 3]
```diff
@@ -1673,9 +1673,12 @@ def upload(pkg_name = nil)
         end
       end
       # Use rubocop to sanitize package file, and let errors get flagged.
-      system 'yes | crew install ruby_rubocop' unless @device[:installed_packages].any? { |elem| elem[:name] == 'ruby_rubocop' }
-      puts "Using rubocop to sanitize #{pkg_file} .".orange
-      system "rubocop -c #{File.join(CREW_LOCAL_REPO_ROOT, '.rubocop.yml')} -A #{pkg_file}", exception: true
+      if @device[:installed_packages].any? { |elem| elem[:name] == 'ruby_rubocop' }
+        puts "Using rubocop to sanitize #{pkg_file} .".orange
```
Comment (uberhacker) [inline bin/crew:1677]: Extra space?

File: bin/crew [hunk 4]
```diff
@@ -1673,9 +1673,12 @@ def upload(pkg_name = nil)
...
+        system "rubocop -c #{File.join(CREW_LOCAL_REPO_ROOT, '.rubocop.yml')} -A #{pkg_file}", exception: true
+      else
+        puts 'Package '.lightred + 'ruby_rubocop'.orange + " is not installed. Rubocop will not be used to sanitize  #{pkg_file} . 😔 You may try this: ".lightred + 'crew install ruby_rubocop'.lightblue
```
Comment (uberhacker) [inline bin/crew:1680]: Extra space?

File: packages/avahi.rb [hunk 5]
```diff
@@ -1,45 +1,95 @@
...
-    system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
+    system "make DESTDIR=#{CREW_DEST_DIR} install"
+    FileUtils.mv "#{CREW_DEST_PREFIX}/etc/avahi/avahi-daemon.conf", "#{CREW_DEST_PREFIX}/etc/avahi/avahi-daemon.conf.default"
+    FileUtils.mv "#{CREW_DEST_PREFIX}/etc/avahi/hosts", "#{CREW_DEST_PREFIX}/etc/avahi/hosts.default"
+    FileUtils.mv "#{CREW_DEST_PREFIX}/etc/avahi/services/sftp-ssh.service", "#{CREW_DEST_PREFIX}/etc/avahi/services/sftp-ssh.service.disabled"
+    FileUtils.mv "#{CREW_DEST_PREFIX}/etc/avahi/services/ssh.service", "#{CREW_DEST_PREFIX}/etc/avahi/services/ssh.service.disabled"
+    # start avahi-daemon from bash.d, which loads after all of env.d via #{CREW_PREFIX}/etc/profile
+    File.write 'bashd_avahi', <<~BASHDAVAHI_EOF
+      [[ $(pgrep -wc avahi-daemon) > 0 ]] || source #{CREW_PREFIX}/bin/startavahi
+    BASHDAVAHI_EOF
```
Comment (uberhacker) [inline packages/avahi.rb:58]: Add to `def self.build` section?

File: packages/avahi.rb [hunk 6]
```diff
@@ -1,45 +1,95 @@
...
+      echo -e "\n${localip} ${hostname}.local" >> #{CREW_PREFIX}/etc/avahi/hosts
+      echo "Enabling Avahi mDNS/DNS-SD daemon for address $localip ..."
+      mkdir -p #{CREW_PREFIX}/var/log && touch #{CREW_PREFIX}/var/log/avahi.log
+      (sudo LD_LIBRARY_PATH="$LD_LIBRARY_PATH" #{CREW_PREFIX}/sbin/avahi-daemon &> #{CREW_PREFIX}/var/log/avahi.log &)
+    STARTAVAHI_EOF
+    FileUtils.install 'start_avahi', "#{CREW_DEST_PREFIX}/bin/startavahi", mode: 0o755
+    File.write 'stop_avahi', <<~STOP_AVAHI_EOF
+      #!/bin/bash
+      killall -9 avahi-daemon
+    STOP_AVAHI_EOF
```
Comment (uberhacker) [inline packages/avahi.rb:88]: Add to `def self.build` section?

File: packages/distcc.rb [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+    distcc_gcc_targets = %W[c++ c89 c99 cc cpp g++ gcc #{CREW_TGT}-g++ #{CREW_TGT}-gcc #{CREW_TGT}-gcc-#{@gcc_version}]
+    distcc_clang_targets = %W[clang clang++ clang-#{@clang_version} clang++-#{@clang_version}]
+    File.write 'gcc-wrapper', <<~GCC_WRAPPEREOF
+      #!/bin/bash
+      if `which ccache &>/dev/null` ; then#{' '}
+        exec ccache distcc #{CREW_PREFIX}/bin/#{CREW_TGT}-g${0:$[-2]} "$@"
+      else
+        exec distcc #{CREW_PREFIX}/bin/#{CREW_TGT}-g${0:$[-2]} "$@"
+      fi
+    GCC_WRAPPEREOF
```
Comment (uberhacker) [inline packages/distcc.rb:63]: Add to `def self.build` section?

File: packages/distcc.rb [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+    FileUtils.install 'bashd_distccd', "#{CREW_DEST_PREFIX}/etc/bash.d/distccd", mode: 0o644
+    File.write 'env.d_distccd', <<~ENVDDISTCCD_EOF
+      PATH=#{CREW_PREFIX}/lib/distcc/bin:$PATH
+      ALLOWEDNETS='127.0.0.1/8'
+      # DISTCC_VERBOSE=1
+      DISTCC_JOBS=`distcc -j`
+      DISTCC_DIR=#{CREW_PREFIX}/tmp/.distcc
+      mkdir -p $DISTCC_DIR
+      STARTDISTCC='true'
+    ENVDDISTCCD_EOF
```
Comment (uberhacker) [inline packages/distcc.rb:170]: Add to def self.build section?

File: bin/crew (hunk 4 above)
Comment (uberhacker) [reply bin/crew:1680]: I'm wondering if we can make a generic function to do all this something like:
```
system "rubocop -c #{File.join(CREW_LOCAL_REPO_ROOT, '.rubocop.yml')} -A #{pkg_file}", exception: true if package_exists(pkg_file)?
```
Appears to be a good use case for reusable code.

**Review by uberhacker — APPROVED** [review]

**Pull request merged**